.github/workflows/ci.yml

```diff
@@ -31,8 +31,9 @@ jobs:
 
       - name: Run cargo test
         run: |
-          (&>/dev/null cargo run --no-default-features -F storage-mem -- start --log trace --user root --pass root memory &)
-          cargo test --workspace --features protocol-ws,protocol-http,kv-rocksdb
+          cargo build --locked --no-default-features --features storage-mem
+          (&>/dev/null ./target/debug/surreal start --log trace --user root --pass root memory &)
+          cargo test --locked --workspace --features protocol-ws,protocol-http,kv-rocksdb
```

Comment:
@tobiemh I'm using `--locked` here to make sure we catch such issues in the CI workflow rather than wait for it to be caught in the nightly build. A nice side-effect of that is that it sped `cargo build` and `cargo test` up.

Instead of `cargo run` I switched to `cargo build` and running the binary after it was already compiled because I was getting a race condition where `cargo test` was trying to connect to the server before it had started, [resulting in connection refused errors](https://github.com/surrealdb/surrealdb/actions/runs/4665071151/jobs/8258003215#step:5:1936).